lib/src/Uncertainty/Distribution/RandomMixture.cxx

```diff
@@ -2870,6 +2870,15 @@ void RandomMixture::setParameter(const Point & parameter)
 {
   if (parameter.getSize() != getParameter().getSize()) throw InvalidArgumentException(HERE) << "Error: expected " << getParameter().getSize() << " values, got " << parameter.getSize();
   const Scalar w = getWeight();
+  const UnsignedInteger size = distributionCollection_.getSize();
```

Comment:
Beware of the simplification mechanism! If you write `Y = Uniform(1.0, 2.0) + Uniform(2.0, 4.0)` it becomes `Y = Triangular(4.0, 5.0, 6.0)` so you have 3 parameters instead of the 4 initial ones. This point can be mentioned in the documentation instead of trying to catch it in the code.

Reply:
I can use setDistributionCollectionAndWeights with simplify=false so that no further simplification is done